src/EscrowERC20.sol

```diff
@@ -3,88 +3,101 @@
 pragma solidity >=0.7.0 <0.9.0;
 
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
 
-contract EscrowERC20 {
+contract EscrowERC20 is ReentrancyGuard {
     /**
      * All Errors
      */
     error EscrowERC20__TransferFailed(address from, address to, uint256 amount);
-    error EscrowERC20__OnlyPlayerAllowed();
     error EscrowERC20__EscrowNotAvailable();
     error EscrowERC20__AddressMustBeNotZero();
+    error EscrowERC721__OnlyAdminAllowed();
     error EscrowERC20__PlayerContractDoNotHaveTokensTransfered();
 
-    /**
-     * All enums
-     */
-    enum EscrERC20Available {
-        NO,
-        YES
-    }
     /**
      * All state variables
      */
 
-    address public escrAddress;
-    address public playerAddress;
-    address public contractAddress;
+    address public adminAddress;
     uint256 public totalBalance;
 
-    EscrERC20Available public escrAvailable;
+    bool public escrAvailable;
     mapping(address => mapping(address => uint256)) playerToContractToAmount;
     mapping(address => address[]) playerToContracts;
 
     constructor() {
-        escrAddress = address(this);
-        escrAvailable = EscrERC20Available.YES;
+        adminAddress = 0x4cd4df5E4485ffd09345bB5dAC0fcE06Dd00ef07;
```

Comment:
Do not hardcode admin address, instead send it as a argument to the constructor.